src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -2785,13 +2785,22 @@ internal void Delete(object parameters)
 
             var command = new DynamoModel.DeleteModelCommand(Guid.Empty);
             this.ExecuteCommand(command);
+            OnNodeDeleted(this, EventArgs.Empty);
```

Comment:
This command seems to be a general entrance for not only nodes but all graph elements, I would suggest trigger event on a different place to be more accurate, e.g. RemoveAndDisposeNode() at WorkspaceModel.cs

Reply:
Great tip, now only when the targeted parent node is deleted the node autocomplete menu gets dismissed.